### ISSUE TITLE
Understand Validation Loss behavior 

### ISSUE THREAD

**Issue by mdatres** [open-mmlab/mmdetection#4996]
Dear all, 

I have a general question about the shape of my validation loss. I use  Foveabox on my costum dataset. The config file is the following: 
```
dataset_type = 'CocoDataset'
data_root = '/data0/Catchme/dataset/16_03_2021/images/'
classes = ('scafoideus_titanus', 'planococcus_ficus_m')
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
albu_train_transforms = [
    dict(type='ShiftScaleRotate', 
        rotate_limit=90,
    ),
    dict(type='RandomSizedBBoxSafeCrop', 
        height = 1152,
        width = 1152,
        p=0.3,
    ),
    dict(
        type='MultiplicativeNoise',
        multiplier=[0.5, 1.5],
        per_channel=True,
        p=0.5),
    dict(
        type='RandomBrightnessContrast',
        p=0.1),
    dict(
        type='JpegCompression',
        quality_lower=39, 
        quality_upper=40, 
       p=0.5),
    dict(type='Cutout',
        num_holes=50,
        max_h_size=20,
        max_w_size=20,
        fill_value=128,
        p=0.1),
    dict(type='Blur', blur_limit=(15,15), p=0.1),
]
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='Resize', img_scale=(1152, 1152), keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(type='Pad', size_divisor=32),
    dict(
        type='Albu',
        transforms=albu_train_transforms,
        bbox_params=dict(
        type='BboxParams',
        format='pascal_voc',
        label_fields=['gt_labels'],
        min_visibility=0.0,
        filter_lost_elements=True),
        keymap={
            'img': 'image',
            'gt_masks': 'masks',
            'gt_bboxes': 'bboxes'
        },
        update_pad_shape=False,
        skip_img_without_anno=True),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(4608, 3456),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=3,
    workers_per_gpu=1,
    train=dict(
        type='CocoDataset',
        ann_file=
        '/home/datres/Catchme/mmdetection/annotations/16-03-21/train.json',
        img_prefix='/data0/Catchme/dataset/16_03_2021/images/Patch/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations', with_bbox=True),
            dict(type='Resize', img_scale=(1152, 1152), keep_ratio=True),
            dict(type='RandomFlip', flip_ratio=0.5),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size_divisor=32),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
        ]),
    val=dict(
        type='CocoDataset',
        ann_file=
        '/home/datres/Catchme/mmdetection/annotations/16-03-21/validation.json',
        img_prefix='/data0/Catchme/dataset/16_03_2021/images/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(4608, 3456),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),
    test=dict(
        type='CocoDataset',
        ann_file=
        '/home/datres/Catchme/mmdetection/annotations/16-03-21/test.json',
        img_prefix='/data0/Catchme/dataset/16_03_2021/images/',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(4608, 3456),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]))
evaluation = dict(interval=1, metric='bbox')
optimizer_config = dict(grad_clip=None)
lr_config = dict(
    policy='step',
    warmup='linear',
    warmup_iters=2000,
    warmup_ratio=0.001,
    step=[16, 22])
runner = dict(type='EpochBasedRunner', max_epochs=50)
checkpoint_config = dict(interval=1)
log_config = dict(
    interval=50,
    hooks=[dict(type='TextLoggerHook'),
           dict(type='TensorboardLoggerHook')])
custom_hooks = [dict(type='NumClassCheckHook')]
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = 'http://download.openmmlab.com/mmdetection/v2.0/foveabox/fovea_r101_fpn_4x4_2x_coco/fovea_r101_fpn_4x4_2x_coco_20200208-02320ea4.pth'
resume_from = None
workflow = [('train', 1), ('val', 1)]
model = dict(
    type='FOVEA',
    pretrained='torchvision://resnet101',
    backbone=dict(
        type='ResNet',
        depth=101,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        norm_cfg=dict(type='BN', requires_grad=True),
        norm_eval=True,
        style='pytorch'),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        start_level=1,
        num_outs=5,
        add_extra_convs='on_input'),
    bbox_head=dict(
        type='FoveaHead',
        num_classes=2,
        in_channels=256,
        stacked_convs=4,
        feat_channels=256,
        strides=[8, 16, 32, 64, 128],
        base_edge_list=[16, 32, 64, 128, 256],
        scale_ranges=((1, 64), (32, 128), (64, 256), (128, 512), (256, 2048)),
        sigma=0.4,
        with_deform=False,
        loss_cls=dict(
            type='FocalLoss',
            use_sigmoid=True,
            gamma=1.5,
            alpha=0.4,
            loss_weight=1.0),
        loss_bbox=dict(type='SmoothL1Loss', beta=0.11, loss_weight=1.0)),
    train_cfg=dict(),
    test_cfg=dict(
        nms_pre=1000,
        score_thr=0.5,
        nms=dict(type='nms', iou_threshold=0.5),
        max_per_img=100))
optimizer = dict(type='Adam', lr=0.0001, weight_decay=0.0001)
work_dir = '/data0/work_dirs/FoveaBox/'
gpu_ids = [0]
```
During the training phase the training loss and the regression validation loss decrease , while the classification validation loss increases (to better explain this behavior I attach the tensorboard's plots). Moreover, notice that, even if the classification validation loss increases, the performance of the model (in metrics' term) seems to increase. I have tried to load the model's weights from your .pth file, and the situation is still worse, i.e. in that case also the regression validation loss increases. Even in this case the metrics become better and better. Is the model overfit? Can you suggest me what happens? What should I do in this case? 

Thanks in advance, 
mdatres
![Schermata 2021-04-17 alle 14 35 26](https://user-images.githubusercontent.com/80750571/115113537-2aaa3380-9f8b-11eb-8c93-98e58b5fb137.png)
![Schermata 2021-04-17 alle 14 35 41](https://user-images.githubusercontent.com/80750571/115113544-30a01480-9f8b-11eb-94af-e05cba59c369.png)

**Comment by ZwwWayne** [comment]
Since you are running the model on the validation set through val workflow, the model is not trained nor updated. So the loss curves only reveal how models behave on the val set, but their increasing/decreasing indicate nothing because they are not trained on val set and the model is not updated..
